server/src/main/java/org/apache/iotdb/db/engine/compaction/no/NoCompactionTsFileManagement.java

```diff
@@ -105,44 +138,61 @@ public void addRecover(TsFileResource tsFileResource, boolean sequence) {
 
   @Override
   public void addAll(List<TsFileResource> tsFileResourceList, boolean sequence) {
-    if (sequence) {
-      sequenceFileTreeSet.addAll(tsFileResourceList);
-    } else {
-      unSequenceFileList.addAll(tsFileResourceList);
+    for (TsFileResource tsFileResource : tsFileResourceList) {
+      add(tsFileResource, sequence);
     }
   }
 
   @Override
   public boolean contains(TsFileResource tsFileResource, boolean sequence) {
     if (sequence) {
-      return sequenceFileTreeSet.contains(tsFileResource);
+      return sequenceFileTreeSetMap
+          .computeIfAbsent(tsFileResource.getTimePartition(), this::newSequenceTsFileResources)
+          .contains(tsFileResource);
```

Comment:
It may be better to use `getOrDefault(partition, Collections.emptyList())` here since it is not necessary to create entries for reads.